src/mcts/node.cc

```diff
@@ -382,6 +382,10 @@ void NodeTree::MakeMove(Move move) {
   }
   gNodePool.ReleaseAllChildrenExceptOne(current_head_, new_head);
   current_head_ = new_head ? new_head : current_head_->CreateChild(move);
+  if (no_tree_reuse) {
+    gNodePool.ReleaseChildren(current_head_);
+    current_head_->ResetStats();
+  }
```

Comment:
Think this function can be written cleaner. There should only be one Release() call for either branch